include/Engine/Utility/Config/ConfigSection.h

```diff
@@ -40,6 +41,43 @@ namespace gp1 {
 			// Get T value of a config.
 			template <typename T>
 			T GetConfigTyped(const std::string& key, T def);
+			// Sets a config value with an enum value name.
+			template <typename T>
+			void SetConfigEnum(const std::string& key, T value, const std::vector<std::pair<T, std::string>>& enumNames) {
+				auto itr = enumNames.begin();
+				while (itr != enumNames.end()) {
+					if (itr->first == value) {
+						SetConfig(key, itr->second);
+						break;
+					}
+					itr++;
+				}
+			}
+
+			// Gets the element T that the config's value was the same as.
+			template <typename T>
+			T GetConfigEnum(const std::string& key, T def, const std::vector<std::pair<T, std::string>>& enumNames) {
+				std::string defName = "";
```

Comment:
Just using `itr->second` directly should work just as well, right? No need for a temporary copy!

Reply:
Maybe I guess I'll try to remove the temp copy.